cli/loader_lib.c

```diff
@@ -198,6 +198,11 @@ __attribute__((constructor)) void jl_load_libjulia_internal(void) {
 
     // Unpack our special library names.  This is why ordering of library names matters.
     libjulia_internal = load_library(special_library_names[0], lib_dir, 1);
+
+    void (*pZN4llvm2cl22ResetCommandLineParserEv)(void) = (void (*)(void))lookup_symbol(libjulia_internal, "_ZN4llvm2cl22ResetCommandLineParserEv");
+    if (pZN4llvm2cl22ResetCommandLineParserEv)
+       pZN4llvm2cl22ResetCommandLineParserEv();
```

Comment:
This is rather gross so I wonder if it makes sense to put it behind `ifdef`? It at least clarifies when it's needed.

```suggestion
#if defined(_COMPILER_ASAN_ENABLED_) || defined(_COMPILER_TSAN_ENABLED_)
    void (*pZN4llvm2cl22ResetCommandLineParserEv)(void) = (void (*)(void))lookup_symbol(libjulia_internal, "_ZN4llvm2cl22ResetCommandLineParserEv");
    if (pZN4llvm2cl22ResetCommandLineParserEv)
       pZN4llvm2cl22ResetCommandLineParserEv();
#endif
```

Reply:
@vtjnash thoughts?

Reply:
For some reason, I had thought the need for this had went away with LLVM13. Is that not true? My testing might have been flawed.

Reply:
Let's see. I pushed a revert and CI should tell us if we still need it.

Reply:
Looks like it sometimes works now